packages/ui/src/components/MenuV2/index.tsx

```diff
@@ -0,0 +1,165 @@
+  | (ReactElement<ButtonHTMLAttributes<HTMLButtonElement>> & {
+      ref?: Ref<HTMLButtonElement>
+    })
+
+const StyledPopup = styled(Popup)`
+  background-color: ${({ theme }) => theme.colors.neutral.background};
+  box-shadow: ${({ theme }) => theme.shadows.menu};
+  padding: 0;
+
+  &[data-has-arrow='true'] {
+    &::after {
+      border-color: ${({ theme }) => theme.colors.neutral.background}
+        transparent transparent transparent;
+    }
+  }
+`
+
+type MenuProps = {
+  id?: string
+  ariaLabel?: string
+  placement?: ComponentProps<typeof Popup>['placement']
+  children?: (() => ReactNode) | ReactNode
+  className?: string
+  disclosure: DisclosureElement
+  hasArrow?: boolean
+  visible?: boolean
+  'data-testid'?: string
+}
+
+const MenuList = styled(Stack)`
+  &:after,
+  &:before {
+    border: solid transparent;
+    border-width: 9px;
+    content: ' ';
+    height: 0;
+    width: 0;
+    position: absolute;
+    pointer-events: none;
+  }
+
+  &:after {
+    border-color: transparent;
+  }
+  &:before {
+    border-color: transparent;
+  }
+  background-color: ${({ theme }) =>
+    theme.colors.neutral.backgroundWeakElevated};
+  color: ${({ theme }) => theme.colors.neutral.text};
+  border-radius: ${({ theme }) => theme.radii.default};
+  position: relative;
+`
+
+const FwdMenu = forwardRef(
+  (
+    {
+      id,
+      ariaLabel = 'Menu',
+      children,
+      disclosure,
+      hasArrow = true,
+      placement = 'bottom',
+      visible = false,
+      className,
+      'data-testid': dataTestId,
+    }: MenuProps,
+    ref: Ref<HTMLButtonElement | null>,
+  ) => {
+    const [isVisible, setIsVisible] = useState(visible)
+    const popupRef = useRef<HTMLDivElement>(null)
+    const disclosureRef = useRef<HTMLDivElement>(null)
+    const tempId = useId()
+    const finalId = `menu-${id ?? tempId}`
+
+    // if you need dialog inside your component, use function, otherwise component is fine
+    const target = isValidElement<ButtonHTMLAttributes<HTMLButtonElement>>(
+      disclosure,
+    )
+      ? disclosure
+      : disclosure({ visible: isVisible })
+    const innerRef = useRef(target as unknown as HTMLButtonElement)
+    useImperativeHandle(ref, () => innerRef.current)
+
+    const toggleVisible = () => {
+      setIsVisible(!isVisible)
+
+      // Focus the first item when the menu is opened
+      if (!isVisible) {
+        requestIdleCallback(() => {
+          // We have to wait for the popup to be inserted in the DOM
+          if (popupRef.current?.firstChild?.firstChild instanceof HTMLElement) {
+            popupRef.current.firstChild.firstChild.focus()
+          }
+        })
+      }
+    }
+
+    const onClose = () => {
+      setIsVisible(false)
+
+      // Focus the disclosure when the menu is closed
+      disclosureRef.current?.focus()
+    }
+
+    const finalDisclosure = cloneElement(target, {
+      onClick: toggleVisible,
+      'aria-haspopup': 'dialog',
```

Comment:
```suggestion
      'aria-haspopup': 'menu',
```

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup

Reply:
That's not what reakit uses:

<img width="806" alt="Screenshot 2023-09-18 at 09 38 27" src="https://github.com/scaleway/ultraviolet/assets/43268759/c6bc21e0-594b-429b-ab07-42e822b355f4">